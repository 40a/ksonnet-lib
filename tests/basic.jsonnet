local v1 = (import "../kube/core.libsonnet").v1;
local v1beta1 = (import "../kube/core.libsonnet").extensions.v1beta1;

{
  containerPortTest1:
    v1.port.container.Default(9090),
  containerPortTest2:
    v1.port.container.Named("containerPort1", 9091),
  containerPortTest3:
    v1.port.container.Default(9092) +
    v1.port.container.Name("containerPort2") +
    v1.port.container.Protocol("UDP") +
    v1.port.container.HostPort(9093) +
    v1.port.container.HostIp("127.0.0.2"),
  deploymentTest:
    local deployment = v1beta1.deployment;
    deployment.Default("hello", {}) +
    deployment.mixin.spec.Selector({frog: "ribbit"}) +
    deployment.mixin.spec.MinReadySeconds(3) +
    deployment.mixin.spec.RollingUpdateStrategy(),
  metadataTest:
    v1.metadata.Default() +
    v1.metadata.Annotations({ annotation1: "label1" }) +
    v1.metadata.Annotations({ annotation2: "label2" }) +
    v1.metadata.Annotation("annotation3", "label3") +
    v1.metadata.Name("name1") +
    v1.metadata.Labels({label1: "value1"}) +
    v1.metadata.Labels({label2: "value2"}) +
    v1.metadata.Label("label3", "value3") +
    v1.metadata.Namespace("namespace1"),
  namespaceTest:
    v1.namespace.Default("namespace1"),
  servicePortTest1:
    v1.port.service.Default(8080),
  servicePortTest2:
    v1.port.service.WithTarget(8081, 8082),
  servicePortTest3:
    v1.port.service.Named("servicePort1", 8083, 8084),
  servicePortTest4:
    v1.port.service.Default(8085) +
    v1.port.service.Name("servicePort2") +
    v1.port.service.Protocol("TCP") +
    v1.port.service.TargetPort(8086) +
    v1.port.service.NodePort(8087),
  serviceTest1:
    v1.service.Default("service1", []),
}