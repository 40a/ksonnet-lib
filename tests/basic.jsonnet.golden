{
   "containerPortTest1": {
      "containerPort": 9090
   },
   "containerPortTest2": {
      "containerPort": 9091,
      "name": "containerPort1"
   },
   "containerPortTest3": {
      "containerPort": 9092,
      "hostIP": "127.0.0.2",
      "hostPort": 9093,
      "name": "containerPort2",
      "protocol": "UDP"
   },
   "deploymentTest": {
      "apiVersion": "extensions/v1beta1",
      "kind": "Deployment",
      "metadata": {
         "annotations": { },
         "labels": { },
         "name": "hello"
      },
      "spec": {
         "minReadySeconds": 3,
         "selector": {
            "matchLabels": {
               "frog": "ribbit"
            }
         },
         "strategy": {
            "rollingUpdate": {
               "maxSurge": 1,
               "maxUnavailable": 1
            },
            "type": "RollingUpdate"
         }
      }
   },
   "metadataTest": {
      "annotations": {
         "annotation1": "label1",
         "annotation2": "label2",
         "annotation3": "label3"
      },
      "labels": {
         "label1": "value1",
         "label2": "value2",
         "label3": "value3"
      },
      "name": "name1",
      "namespace": "namespace1"
   },
   "namespaceTest": {
      "apiVersion": "v1",
      "kind": "Namespace",
      "metadata": {
         "annotations": { },
         "labels": { },
         "name": "namespace1"
      }
   },
   "servicePortTest1": {
      "port": 8080
   },
   "servicePortTest2": {
      "port": 8081,
      "targetPort": 8082
   },
   "servicePortTest3": {
      "name": "servicePort1",
      "port": 8083,
      "targetPort": 8084
   },
   "servicePortTest4": {
      "name": "servicePort2",
      "nodePort": 8087,
      "port": 8085,
      "protocol": "TCP",
      "targetPort": 8086
   },
   "serviceTest1": {
      "apiVersion": "v1",
      "kind": "Service",
      "metadata": {
         "annotations": { },
         "labels": { },
         "name": "service1"
      },
      "spec": {
         "ports": [ ]
      }
   }
}
